tests/test_highlighting/highlighter.py

```diff
@@ -0,0 +1,157 @@
+
+#######################################################################################
+##                     This part concerns the simulation by Ace                      ##
+#######################################################################################
+
+
+import re
+
+# Transcription of the used tokens into a symbol (a letter) in order to apply a coloring
+Tokencode = {
+    "text" :               'T',
+    "keyword" :            'K',
+    "comment" :            'C',
+    "variable" :           'N',
+    "constant.character" : 'S',
+    "invalid" :            'I',
+}
+
+
+
+# This function allows to check 2 syntactic colorations one desired
+# and the other obtained, and manages some special cases
+
+# Returns a boolean and a value:
+# - If the 2 highlights are consistent, the boolean will be true and the value unused
+# - if there is an error in the highlight, the boolean will be false and
+#   the value will be the number of the 1st character where there is an inconsistency
+def check(result,expected):
+    if len(result) != len(expected):
+        raise ValueError("The desired highlight and the obtained highlight do not have the same length !")
```

Comment:
This stops all tests completely. Shouldn't it just be a type of error that prints examples of the wrong values?

Reply:
Normally not, since expected is generated from the code, which normally has the same size as Result, if we get to this Raise, it's either that the test script doesn't work, or that the tests are inconsistent.

Reply:
> Normally not, since expected is generated from the code

I'm not sure I follow. From the code below:

```py
for test in dataTests:
    # get the others data for the tests
    code = test['code']
    expected = test['result']

    # simulation and comparison of the result
    result = simulateRulesWithoutToken(Rules,code)
    valid, indError = check(result,expected)
```

It looks like `expected` comes from the JSON, and so is whatever a human types in there?